spec/docs/expressions.md

```diff
@@ -0,0 +1,62 @@
+# Generating values with expressions
+
+<dfn>Expressions</dfn> are mapping constructs that can be evaluated on a [=logical iteration=], according to the specified reference formulation, to generate values during the mapping process.
+
+## Expression map (`rml:ExpressionMap`)
+
+An <dfn>expression map</dfn> (`rml:ExpressionMap`) is an abstract class, that is specialized by other RML classes. An [=expression map=] MAY have one the following properties:
+* 0 or 1 `rml:constant`, or
+* 0 or 1 `rml:reference`, or
+* 0 or 1 `rml:template`.
+
+Each of these properties specifies an [=expression=] which, upon evaluation, results in a set of values.
+
+The <dfn>reference expression set</dfn> of an [=expression map=] ar the set of expressions which are evaluated on a [=logical iteration=].
+
+### Constant expression (`rml:constant`)
+
+A <dfn>constant-valued expression map</dfn> is an [=expression map=] that always generates the same expression value. A constant-valued expression map is represented by a resource that has exactly one `rml:constant` property, the value of which is called a <dfn>constant expression</dfn>.
+
+The <dfn>constant value</dfn> is a singleton set containing the [=constant expression=].
+
+The [=reference expressions=] of a [constant-valued expression map=] is the empty set.
+
+### Reference (`rml:reference`)
+A <dfn>reference-valued expression map</dfn> is an [=expression map=] that is represented by a resource that has exactly one `rml:reference` property, the value of which is called a <dfn>reference expression</dfn>.
+
+The [=reference expression=] MUST be a valid [=expression=] according to the defined [=reference formulation=] in the [=logical source=].
+
+The [=reference expression set=] of a [reference-valued expression map=] is the singleton set containing the [=reference expression=].
+
+The <dfn>reference value</dfn> is a set of values obtained by evaluating the [=reference expression=] against a given [=logical iteration=].
+
+### Template (`rml:template`)
```

Comment:
We should define what should happen if a parameter is missing from the template. I would say that this returns `null` (or something like that).